cluster/log-dump/log-dump.sh

```diff
@@ -87,7 +87,7 @@ function setup() {
 }
 
 function log-dump-ssh() {
-  if [[ "${gcloud_supported_providers}" =~ "${KUBERNETES_PROVIDER}" ]]; then
+  if [[ "${gcloud_supported_providers}" =~ ${KUBERNETES_PROVIDER} ]]; then
```

Comment:
we're _removing_ quotes??

Reply:
Thanks for the response. We can remove from the below.
```
In ./cluster/log-dump/log-dump.sh line 90:
  if [[ "${gcloud_supported_providers}" =~ "${KUBERNETES_PROVIDER}" ]]; then
                                           ^----------------------^ SC2076: Don't quote right-hand side of =~, it'll match literally rather than as a regex.
```

I've checked the behavior using the following sample script.
```
#!/bin/bash

readonly gcloud_supported_providers="gce gke kubernetes-anywhere"
if [[ "${gcloud_supported_providers}" =~ ${KUBERNETES_PROVIDER} ]]; then
  echo "ok"
fi
```

Reply:
I see. this check is a bit of a hack, what we're actually trying to do is see if KUBERNETES_PROVIDER is in a set, but what we're doing is a regex match :this-is-fine:

this change seems reasonable